### ISSUE TITLE
Unhandled promise rejection (id: 0) on Android back button after immediatelyResetStack

### ISSUE THREAD

**Issue by ferrannp** [exponentjs/ex-navigation#277]
I have routes like:

- Login -> after login I call `immediatelyResetStack` and point to Home route
- Home has Drawer + Tabs
- In the second Tab I have a log out button that `immediatelyResetStack` to Login

All seems to work fine but when I click the back button on Android, after logging out (so in Login page) I get the following error:

```
Unhandled promise rejection (id: 0)
undefined is not an object (evaluating 'currentNavigationStack.index')
```

Expected behaviour would be the app to close.

**Comment by junshengpierre** [comment]
I'm experiencing the exact same issue.

**Comment by shercoder** [comment]
@ferrannp Have you had any luck with finding the solution for this? I am in similar situation. Back button works within the app but when you try to exit out of the app it does not leave the app.

**Comment by ferrannp** [comment]
Not really, just moving to https://github.com/react-community/react-navigation.

**Comment by fadeltd** [comment]
I'm having exact same issue, I can exit from app after logout and reset the stack in android with BackHandler, 
  ```
  BackHandler.addEventListener('hardwareBackPress', function () {
    BackHandler.exitApp();
  });
  ```
But when I got back to the app with android menu button, the warning still exists.
Still no fix until now?
